modules/travel_pay/app/services/travel_pay/service.rb

```diff
@@ -2,20 +2,38 @@
 
 module TravelPay
   class Service
-    def get_claims(current_user)
-      claims_response = client.get_claims(current_user)
-      symbolized_body = claims_response.body.deep_symbolize_keys
+    def get_claims(current_user, params = {})
+      faraday_response = client.get_claims(current_user)
+      raw_claims = faraday_response.body['data'].deep_dup
+
+      claims = filter_by_date(params['appt_datetime'], raw_claims)
 
       {
-        data: symbolized_body[:data].map do |sc|
-          sc[:claimStatus] = sc[:claimStatus].underscore.titleize
+        data: claims.map do |sc|
+          sc['claimStatus'] = sc['claimStatus'].underscore.titleize
           sc
         end
       }
     end
 
     private
 
+    def filter_by_date(date_string, claims)
+      if date_string.present?
+        parsed_appt_date = Date.parse(date_string)
+
+        claims.filter do |claim|
+          !claim['appointmentDateTime'].nil? &&
+            parsed_appt_date == Date.parse(claim['appointmentDateTime'])
```

Comment:
What if the claim doesn't have an `appointmentDateTime`? First, is it even possible that a claim can be created without an appointment date attached to it, and if it is possible will it always be returned regardless of if it matches the params? Or only in the "return all claims" cases?

Reply:
> is it even possible that a claim can be created without an appointment date attached to it

Yeah unfortunately this is a case we had to handle in another part of the app.

Reply:
The claims without appointment datetimes will only be returned in the "return all claims" cases. I thought about trying to cover the case where you search by appt date and somehow the claim w/o an appt date is returned, but there's no way I could see to do it cleanly.

Reply:
Fair. I guess the use case for searching by appointment date would be in order to match against an appointment, so it shouldn't matter.... unless the claim _is_ for that appointment, and maybe has the appointment ID associated with it, but we won't find it if it doesn't have the date attached.... 🤔 (Might be a weird edge case that is not relevant though, I'm not sure how common it is.)